verification/verify.go

```diff
@@ -153,3 +160,75 @@ func IsRetryable(err error) bool {
 	_, retryable := err.(Retryable)
 	return retryable
 }
+
+func verifyPixelParams(params *Params) error {
+	// check if the node is on-chain mode
+	if params != nil &&
+		params.Orchestrator != nil &&
+		params.Orchestrator.TicketParams != nil {
+
+		var err error
+		for i := 0; err == nil && i < len(params.Results.Segments); i++ {
+			rendition := []byte{}
+			if i < len(params.Renditions) {
+				rendition = params.Renditions[i]
+			}
+			if err := verifyPixels(
+				params.Results.Segments[i].Url,
+				rendition,
+				params.Results.Segments[i].Pixels); err != nil {
+
+				glog.Error(err)
+				return ErrPixelMismatch
```

Comment:
Is there a particular reason for dropping the returned error here and using `ErrPixelMismatch` ?

One side effect that comes to mind is that `ErrPixelMismatch` is marked as retryable (eg, re-transcoding with another O might fix the problem), but most of the other potential errors from `verifyPixels` are IO-related. It's less likely that a retry would fix IO errors. For example, IO errors are not retried in the [epic classifier](https://github.com/livepeer/go-livepeer/blob/0670fe4097f2d7312bdc818f2aa5d041ec565b80/verification/epic.go).

Also if we propagate the error from verifyPixels then we probably wouldn't need to print it or import `glog`. (It was kind of neat not having to import any logging here, but that probably can't last.)

Reply:
We have to return `ErrPixelMismatch` here in order to pass the existing tests. We can change it back, along with the tests, but I'm not sure if that's what we want.  I am indifferent about this - maybe @yondonfu can weigh in?

Reply:
I don't think we want to retry on I/O errors so I think it makes sense to propagate the error and to adjust the tests.

Reply:
done in 458cfcf64197302e40ac39ab4511f4218f8f3740 & 86718ed9956dd3536ee4296aab48c40b05982e98